src/jsMain/kotlin/io/beatmaps/maps/mapEmbed.kt

```diff
@@ -0,0 +1,49 @@
+package io.beatmaps.maps
+
+import external.axiosGet
+import io.beatmaps.Config
+import io.beatmaps.History
+import io.beatmaps.api.MapDetail
+import io.beatmaps.index.beatmapInfo
+import io.beatmaps.util.useAudio
+import react.Props
+import react.dom.div
+import react.fc
+import react.router.useNavigate
+import react.router.useParams
+import react.useEffect
+import react.useState
+
+val mapEmbed = fc<Props> {
+    val (map, setMap) = useState<MapDetail?>(null)
+
+    val params = useParams()
+    val history = History(useNavigate())
+    val audio = useAudio()
+
+    fun loadMap() {
+        val mapKey = params["mapKey"]
+
+        setMap(null)
+
+        axiosGet<MapDetail>(
+            "${Config.apibase}/maps/id/$mapKey"
+        ).then {
+            setMap(it.data)
+        }.catch {
+            history.push("/")
```

Comment:
I really don't think you want to do this. Need some fancy error message instead.

The header isn't there and it's going to be weird having the homepage in an iframe...
<img width="1226" alt="Screenshot 2024-06-24 at 15 31 16" src="https://github.com/beatmaps-io/beatsaver-main/assets/534069/14f61c4c-2971-48f9-8206-e2fd5ce8f917">

Reply:
Oh you're right, I'll see what I can do to fix that.